### PR TITLE
Add default delete action to actions dropdown in object detail modal

### LINKED ISSUE #32323: Add default delete action to actions dropdown in object detail modal
Epic: #32320
Product doc: [link](https://www.notion.so/Allow-basic-deletes-and-updates-on-object-detail-views-f05b298c3aba4a6ca883b2738d86824d)
Depends on: #32321
Related to: #32322

----

- Clicking delete should open a confirmation modal where the user has to confirm that they wish to delete the record and that there’s no undo.
- If a successful delete is run from an object detail modal, the detail modal should close and the user should see a toast. The row that was deleted should no longer be present in the table.

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #32323

### Description

- Tests will be added in #32350

### How to verify

> Clicking delete should open a confirmation modal where the user has to confirm that they wish to delete the record and that there’s no undo.

> If a successful delete is run from an object detail modal, the detail modal should close and the user should see a toast. The row that was deleted should no longer be present in the table.

### Design notes
- The object detail modal has up/down arrows to navigate between neighboring rows. We don't need to close the object detail modal after deleting the object - we can navigate to next or previous object, or close the modal if there are no more objects. I believe [this was the original intention](https://www.notion.so/metabase/Allow-updates-and-deletes-on-model-object-detail-views-f05b298c3aba4a6ca883b2738d86824d?pvs=4#1ff1b9c44cd544d5b60881f80bcf272f) for the descoped part with objects displayed in dashboards. This PR implements closing both modals when deleting objects. If you want to try out how it would have worked with auto-navigation to sibling row please `git cherry-pick 8b24685eee5cd968f53cc14b7194a66588fb5845` into this branch
- Deleting an object that is referenced by other objects will often result in an error ("Referential integrity constraint violation") - see the end of demo - we might want to check if it's possible to hide/disable deleting in that case beforehand. Or consider cascade delete option?

### Demo

This demo uses 2 simple models that do nothing but reference sample db tables: accounts & reviews respectively.

https://github.com/metabase/metabase/assets/6830683/6b954457-1aa1-41c7-b154-26ffd05baaca

### CODE REVIEW

**Review by kdoh — COMMENTED** [review]
Feeling good in general. A couple things are top of mind for me.

- I'm seeing [a bit of jumpiness](https://www.loom.com/share/a16687a78f9d4acf97803a0c8670ca63) on the update modal when the data loads. It's non blocking, but would be great to fix if it's a matter of when we do the content check.
- Conor and I will get you an answer on cascading deletes. We should determine if we want to block this PR on it or make a note in the Epic to follow up once we know what route we want to go. 
- R.e. the whole move to another record or close the object detail after a successful delete bit, I think closing the modal makes the most sense so let's continue on that route. Thanks for looking into that other approach, your intuition was right. Moving to another record felt more important in the dashboard context so there wasn't a content gap all of the sudden.

**Comment by cdeweyx** [comment]
Looking real nice. Couple of open items before I approve:

> I'm seeing [a bit of jumpiness](https://www.loom.com/share/a16687a78f9d4acf97803a0c8670ca63) on the update modal when the data loads. It's non blocking, but would be great to fix if it's a matter of when we do the content check.

I would love to try and resolve this before merging. It's pretty in-your-face and while it's a split second, it makes me a little hesitant about the consequential thing I'm about to do.

> Conor and I will get you an answer on cascading deletes. We should determine if we want to block this PR on it or make a note in the Epic to follow up once we know what route we want to go.

I agree with Kyle's suggestion [here](https://metaboat.slack.com/archives/C057LE0Q4PQ/p1690295017296989?thread_ts=1690208614.530869&cid=C057LE0Q4PQ): "We’re able to tell beforehand that the current entity is referenced by other tables and add a “disabled” state to the delete option." That seems like the way to go for me. I don't think it has to block this PR but it's a definite follow-up item.

File: frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx
Comment (uladzimirdev) [inline frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx:29]: so, am I right, disabled is only for submit button? can we rename it then?

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx
Comment (uladzimirdev) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx:439]: can't you move `Modal` wrapper into `DeleteObjectModal`?

File: frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx:31]: I just dropped an [RFC about it](https://www.notion.so/metabase/56f6f59da3b647de89acb28ea9ebc424?v=cca33933b6244a7a94f256d1c110ce09&p=5e63b631c8c545a9bbd272e54deeeebe&pm=s).

File: frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx:39]: What's the advantage?

File: frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx:56]: It works differently, if it's an array there will automatically be margins between the buttons:
https://github.com/metabase/metabase/blob/3f35c0d96c8fab4d14bf3aa93464bfa80f801955/frontend/src/metabase/components/ModalContent/ModalContent.tsx#L191-L197

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx:439]: I wanted to be consistent with the current code:
- the `ActionExecuteModal` in lines above it
- there are lots (dozens) of components called `XYZModal` that render `<ModalContent>` as their root DOM element (examples: `CollectionMoveModal`, `AddToDashSelectDashModal`, `QuestionSavedModal`)

File: frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx
Comment (uladzimirdev) [reply frontend/src/metabase/visualizations/components/ObjectDetail/DeleteObjectModal.tsx:56]: oh, I didn't know that, thanks!

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx
Comment (uladzimirdev) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx:439]: it's not straightforward, but I agree with your points

**Review by uladzimirdev — APPROVED** [review]

**Comment by uladzimirdev** [comment]
LGTM

**Pull request merged**
